encoding/thrift/outbound_nowire.go

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package thrift
+
+import (
+	"bytes"
+	"context"
+
+	"go.uber.org/thriftrw/protocol"
+	"go.uber.org/thriftrw/protocol/stream"
+	"go.uber.org/thriftrw/wire"
+	"go.uber.org/yarpc"
+	encodingapi "go.uber.org/yarpc/api/encoding"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/encoding/thrift/internal"
+	"go.uber.org/yarpc/pkg/encoding"
+	"go.uber.org/yarpc/pkg/errors"
+	"go.uber.org/yarpc/pkg/procedure"
+)
+
+// NoWireClient is a generic Thrift client for encoding/decoding using
+// ThriftRW's "streaming" mechanisms.
+// It speaks in raw Thrift payloads.
+//
+// Users should use the client generated by the code generator rather than
+// using this directly.
+type NoWireClient interface {
+	// Call the given Thrift method.
+	Call(ctx context.Context, reqBody stream.Enveloper, resBody stream.BodyReader, opts ...yarpc.CallOption) error
+	CallOneway(ctx context.Context, reqBody stream.Enveloper, opts ...yarpc.CallOption) (transport.Ack, error)
+
+	// Enabled returns whether or not this client is enabled through a
+	// ClientOption.
+	Enabled() bool
```

Comment:
I think I don't have all of the context here, but what is the scenario where a `NoWireClient` would not be enable to use no wire? Isn't it the concept of having a NoWire client and a WithWire client?

Reply:
Yes, the idea is that the client and server are agnostic to whether or not the other side is using the "no-wire" implementation.  There shouldn't exist a scenario where this matters.

`Enabled()` exists to allow the clients to opt-in to using the "no-wire" implementation, for simplifying onboarding.